tutorials/introduction-to-jsosolvers/index.jmd

```diff
@@ -0,0 +1,132 @@
+---
+title: "Introduction to JSOSolvers"
+tags: ["solvers", "nlpmodels", "models"]
+author: "Tangi Migot"
+---
+
+\toc
+
+# JSOSolvers.jl Tutorial
+
+This package provides optimization solvers curated by the
+[JuliaSmoothOptimizers](https://juliasmoothoptimizers.github.io)
+organization.
+All solvers are based on [NLPModels.jl](https://github.com/JuliaSmoothOptimizers/NLPModels.jl) and [SolverCore.jl](https://github.com/JuliaSmoothOptimizers/SolverCore.jl).
+
+This package contains the implementation of four algorithms that are classical for unconstrained/bound-constrained nonlinear optimization:
+`lbfgs`, `R2`, `tron`, and `trunk`.
+
+## Solver input and output
+
+All solvers have the following signature:
+
+```
+    stats = name_solver(nlp; kwargs...)
+```
+
+where `name_solver` can be `lbfgs`, `R2`, `tron`, or `trunk`, and with
+- `nlp::AbstractNLPModel{T, V}` is an AbstractNLPModel or some specialization, such as an `AbstractNLSModel`;
+- `stats::GenericExecutionStats{T, V}` is a `GenericExecutionStats`, see `SolverCore.jl`.
+
+The keyword arguments may include:
+- `x::V = nlp.meta.x0`: the initial guess.
+- `atol::T = √eps(T)`: absolute tolerance.
+- `rtol::T = √eps(T)`: relative tolerance, the algorithm stops when ‖∇f(xᵏ)‖ ≤ atol + rtol * ‖∇f(x⁰)‖.
+- `max_eval::Int = -1`: maximum number of objective function evaluations.
+- `max_time::Float64 = 30.0`: maximum time limit in seconds.
+- `verbose::Int = 0`: if > 0, display iteration details every `verbose` iteration.
+
+Refer to the documentation of each solver for further details on the available keyword arguments.
+
+## Specialization for nonlinear least-squares
+
+The solvers `tron` and `trunk` both have a specialized implementation for input models of type `AbstractNLSModel`.
+
+The following example illustrate this specialization.
+
+```julia
+using JSOSolvers, ADNLPModels
+f(x) = (x[1] - 1)^2 + 4 * (x[2] - x[1]^2)^2
+nlp = ADNLPModel(f, [-1.2; 1.0])
+trunk(nlp, verbose = 1)
```

Comment:
```suggestion
trunk(nlp)
```
The `verbose=1` is not visualized, I guess because of the way the logging and building the output of this work.